sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -905,6 +923,7 @@ export class ServiceBusMessageImpl implements ServiceBusReceivedMessage {
     this._rawAmqpMessage = _rawAmqpMessage;
     this._rawAmqpMessage.bodyType = actualBodyType;
     this.delivery = delivery;
+    this.state = restOfMessageProps.state;
```

Comment:
Doesnt the `Object.assign(this, restOfMessageProps);` at line 902 take care of assigning the state?

Reply:
TypeScript compiler is not smart enough to detect that.  I added some comments.